src/CoreLib/DSeq.fs

```diff
@@ -532,71 +537,88 @@ and [<AllowNullLiteral>]
-                Logger.LogF( LogLevel.WildVerbose, ( fun _ -> sprintf "Rcvd command %A from peer %d" cmd peeri ))
-                x.SyncCloseFromPeer queuePeer peeri |> ignore
+                Logger.LogF( jobID, LogLevel.MildVerbose, ( fun _ -> sprintf "Rcvd command %A from peer %d" cmd peeri ))
+                x.SyncCloseFromPeer queuePeer peeri jobID |> ignore
                 true
 
             | _ ->
-                Logger.Log( LogLevel.Info, ( sprintf "DStream.Callback(%s:%s), Unexpected command %A, peer %d" x.Name x.VersionString cmd peeri )    )
+                let msg = sprintf "DStream.Callback(%s:%s), Unexpected command %A, peer %d" x.Name x.VersionString cmd peeri
+                jobAction.ThrowExceptionAtContainer( msg )
                 true
-        with 
-        | e -> 
-            Logger.Log( LogLevel.Info, ( sprintf "Error in processing DStream.ProcessJobCommand(%s:%s) cmd %A, peer %d, with exception %A" x.Name x.VersionString cmd peeri e )    )
-            true
-    member internal x.DStreamCallback( cmd, peeri, ms, name, verNumber ) =
-        try
-            // Outgoing queue for the reply 
-            let q = x.Cluster.Queue( peeri )
-            match ( cmd.Verb, cmd.Noun ) with 
-            | ( ControllerVerb.Echo, ControllerNoun.DStream ) ->
-                ()
-            | ( ControllerVerb.ConfirmClosePartition, ControllerNoun.DStream ) ->
-                ()
-            | ( ControllerVerb.ConfirmClose, ControllerNoun.DStream ) ->
-                x.bSentPeerCloseConfirmed.[peeri] <- true
-                Logger.Log( LogLevel.MildVerbose, ( sprintf "DStream %s:%s, ConfirmClose received from peer %d" x.Name x.VersionString peeri )    )
-                x.CheckAllPeerClosed()
-            | _ ->
-                Logger.Log( LogLevel.Info, ( sprintf "DStream.Callback(%s:%s), Unexpected command %A, peer %d" x.Name x.VersionString cmd peeri )    )
-        with 
-        | e -> 
-            Logger.Log( LogLevel.Info, ( sprintf "Error in processing DStream.Callback(%s:%s) cmd %A, peer %d, with exception %A" x.Name x.VersionString cmd peeri e )    )
-        true
+//                with 
+//                | ex -> 
+//                    jobAction.EncounterExceptionAtContainer( ex, sprintf "Error in processing DStream.ProcessJobCommand cmd %A, with payload of %dB" 
+//                                                                        cmd ms.Length )
+//                    true
+//        )
+    /// Parse echoing from DStream from a container to other container
+    member internal x.DStreamCallback( cmd, peeri, ms, jobID, name, verNumber ) =
+        using ( SingleJobActionContainer.TryFind(jobID)) ( fun jobAction -> 
+            if Utils.IsNull jobAction then 
+                Logger.LogF( jobID, LogLevel.Info, ( fun _ -> sprintf "[May be Ok, Job cancelled?] DStream.DStreamCallback, received command %A of payload %dB, but can't find job lifecycle object" 
```

Comment:
"Info" may get us a lot of log entries here.  In case of a DSet with many elements, and user only does a take 10.  It may take a while for all peers realize the cancellation. At the same time, we may get a lot of such log entries?